crates/rpc/rpc/src/reth.rs

```diff
@@ -0,0 +1,148 @@
+use crate::eth::error::{EthApiError, EthResult};
+use async_trait::async_trait;
+use jsonrpsee::core::RpcResult;
+use reth_interfaces::Result;
+use reth_primitives::{Account, Address, BlockId, U256};
+use reth_provider::{AccountChangeReader, BlockReaderIdExt, StateProviderFactory};
+use reth_rpc_api::RethApiServer;
+use reth_tasks::TaskSpawner;
+use std::{collections::HashMap, future::Future, sync::Arc};
+use tokio::sync::oneshot;
+
+/// `reth` API implementation.
+///
+/// This type provides the functionality for handling `reth` prototype RPC requests.
+pub struct RethApi<Provider> {
+    inner: Arc<RethApiInner<Provider>>,
+}
+
+// === impl RethApi ===
+
+impl<Provider> RethApi<Provider> {
+    /// The provider that can interact with the chain.
+    pub fn provider(&self) -> &Provider {
+        &self.inner.provider
+    }
+
+    /// Create a new instance of the [RethApi]
+    pub fn new(provider: Provider, task_spawner: Box<dyn TaskSpawner>) -> Self {
+        let inner = Arc::new(RethApiInner { provider, task_spawner });
+        Self { inner }
+    }
+}
+
+impl<Provider> RethApi<Provider>
+where
+    Provider: AccountChangeReader + BlockReaderIdExt + StateProviderFactory + 'static,
+{
+    /// Executes the future on a new blocking task.
+    async fn on_blocking_task<C, F, R>(&self, c: C) -> EthResult<R>
+    where
+        C: FnOnce(Self) -> F,
+        F: Future<Output = EthResult<R>> + Send + 'static,
+        R: Send + 'static,
+    {
+        let (tx, rx) = oneshot::channel();
+        let this = self.clone();
+        let f = c(this);
+        self.inner.task_spawner.spawn_blocking(Box::pin(async move {
+            let res = f.await;
+            let _ = tx.send(res);
+        }));
+        rx.await.map_err(|_| EthApiError::InternalEthError)?
+    }
+
+    #[allow(unused)]
+    async fn account_changes_in_block(
```

Comment:
I left this in here as an example use case, but of course had to mark it `unused`, should I remove it?  If not I'll add a docstring explaining this.

Reply:
Add a docstring but still let me know if you'd prefer I just remove it.

Reply:
if it is not used, it is best to remove it

Reply:
Removed in [a92b817](https://github.com/paradigmxyz/reth/pull/3768/commits/a92b817cb962addac3d28d83da63023a39da98f0)